imap/jmap_mdn.c

```diff
@@ -454,12 +454,10 @@ static json_t *send_mdn(struct jmap_req *req, struct mdn_t *mdn,
     if (r) {
         const char *desc = smtpclient_get_resp_text(*sm);
 
-        syslog(LOG_ERR, "MDN/send failed: %s", desc ? desc : error_message(r));
+        syslog(LOG_ERR, "MDN/send failed: %s", desc);
 
-        if (desc) {
-            err = json_pack("{s:s, s:s}", "type", "serverFail",
-                            "description", desc);
-        }
+        err = json_pack("{s:s, s:s}", "type", "serverFail",
+                        "description", desc);
```

Comment:
This isn't the right fix.  Even though `desc` will never be NULL, it may point to an empty string `""`.

I think the condition in the ternary should be `desc[0]` rather than `desc`, something like `syslog(..., desc[0] ? desc : error_message(r));`

I'm not sure what the right fix for the `if (...)` block is.  We could make the condition `desc[0]` like the ternary, and therefore only report an error if there is an error string. But even if there is no error string, we know there was an error because of the `if (r)` above.  So it seems strange to maybe not report it.  Maybe this part should also fall back to `error_message(r)` when desc is the empty string.  But I don't know the larger context here -- input from @ksmurchison or @rsto would clear this up.

Reply:
Right after `desc` is assigned, it should read `if (!desc[0]) desc = error_message(r)` or the like.